File: Easy/JavaScript/1710. Maximum Units on a Truck.js
var maximumUnits = function(boxTypes, truckSize) {
    boxTypes.sort((a, b) => b[1] - a[1]);
    let units = 0;

    boxTypes.forEach((box) => {
        let count = Math.min(box[0], truckSize);
        units += count * box[1];
        truckSize -= count;
        if(truckSize == 0)
            return units;
    })

    return units;
};

File: Easy/Java/1710. Maximum Units on a Truck.java
class Solution {
    public int maximumUnits(int[][] boxTypes, int truckSize) {
        Arrays.sort(boxTypes, (a, b) -> b[1] - a[1]);
        int units = 0;

        for(int[] box : boxTypes){
            int count = Math.min(box[0], truckSize);
            units += count * box[1];
            truckSize -= count;
            if(truckSize == 0)
                return units;
        }

        return units;
    }
}
